### PR TITLE
UncaughtExceptionHandler added to thread

### PR DESCRIPTION
**Pull request by djalova**
Developer's Certificate of Origin 1.1

       By making a contribution to this project, I certify that:

       (a) The contribution was created in whole or in part by me and I
           have the right to submit it under the Apache License 2.0; or

       (b) The contribution is based upon previous work that, to the best
           of my knowledge, is covered under an appropriate open source
           license and I have the right under that license to submit that
           work with modifications, whether created in whole or in part
           by me, under the same open source license (unless I am
           permitted to submit under a different license), as indicated
           in the file; or

       (c) The contribution was provided directly to me by some other
           person who certified (a), (b) or (c) and I have not modified
           it.

       (d) I understand and agree that this project and the contribution
           are public and that a record of the contribution (including all
           personal information I submit with it, including my sign-off) is
           maintained indefinitely and may be redistributed consistent with
           this project or the open source license(s) involved.

### CODE REVIEW

File: src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java [hunk 1]
```diff
@@ -117,12 +109,14 @@ public void run() {
...
-            response = client.execute(request);
-            responseCode = response.getStatusLine().getStatusCode();
+            if (responseCode == 401) { // Unauthorized error
+              mAccount.authenticate();
+              request.removeHeaders("X-Auth-Token");
+              request.addHeader("X-Auth-Token", mAccount.getAuthToken());
+              LOG.warn("Token recreated for {}.  Retry request", mUrl.toString());
+              response = client.execute(request);
+              responseCode = response.getStatusLine().getStatusCode();
+            }
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java]: @djalova So what should we do if it's not 401? I guess we need to throw exception?

File: src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java [hunk 2]
```diff
@@ -117,12 +109,14 @@ public void run() {
...
           }
         } catch (IOException e) {
           LOG.error(e.getMessage());
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java]: What happens in this case? We just print it to log and that's it? Try to proceed as usual? I assume that SwiftOutputStream need to throw exception to upper layer, so operations will stop. How to catch that exception when it thrown from the thread?

File: src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java [hunk 3]
```diff
@@ -133,6 +127,13 @@ public void run() {
 
   private void startThread() {
     if (writeThread.getState().equals(Thread.State.NEW)) {
+      Thread.UncaughtExceptionHandler handler = new Thread.UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+          LOG.error(t.getName() + e);
```
Comment (gilv) [inline src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java]: @djalova  the same, i think documenting exception is not enough. We need to make sure that SwiftoutputSteam will throw an exception to upper layers so operations will be stopped.

**Comment by djalova** [comment]
@gilv I added in exceptions for HTTP error codes and also propagated exceptions to the upper layers.

**Comment by gilv** [comment]
@djalova thanks. merged

**Pull request closed without merging**